fem/nonlininteg.hpp

```diff
@@ -72,6 +72,42 @@ class NonlinearFormIntegrator
 };
 
 
+class BlockNonlinearFormIntegrator
+{
+public:
+   /// Compute the local energy
+   virtual double GetElementEnergy(Array<const FiniteElement *>&el,
+                                   ElementTransformation &Tr,
+                                   Array<const Vector *>&elfun);
+
+   /// Perform the local action of the BlockNonlinearFormIntegrator
+   virtual void AssembleElementVector(Array<const FiniteElement *> &el,
+                                      ElementTransformation &Tr,
+                                      Array<Vector *> &elfun,
+                                      Array<Vector *> &elvec);
```

Comment:
The `elfun` arguments should probably be arrays of `const Vector *`.

Reply:
Done. I had to add a new variable (`el_x_const`)[https://github.com/mfem/mfem/blob/7e6b0be6b7e28e835c9376bb18324bacd39d63ed/fem/nonlinearform.cpp#L455] to add the const declaration inside the template parameter. I thought this would be better than using `const_cast`, but let me know if you think otherwise.